clang/test/CIR/Transforms/simpl.cir

```diff
@@ -0,0 +1,41 @@
+// RUN: cir-opt %s -cir-simplify -o - | FileCheck %s
+
+!s32i = !cir.int<s, 32>
+!s64i = !cir.int<s, 64>
+module {
+cir.func @foo(%arg0: !cir.ptr<!s32i>) {
```

Comment:
What's the C code corresponding to this? Add a C code test instead with what you want to optimize.

For a `cir-opt` tests, better to split into two really simple CIR functions, each one testing the specific opt, so it's pretty clear you are adding coverage for different individual peepholes.

Reply:
done!